docs/email/postfix/configure-spf-and-dkim-in-postfix-on-debian-8.md

```diff
@@ -0,0 +1,403 @@
+
+    {: .file-excerpt}
+    /etc/postfix/main.cf
+    :   ~~~ conf
+        # Milter configuration
+        # OpenDKIM
+        milter_default_action = accept
+        milter_protocol = 2
+        smtpd_milters = local:/opendkim/opendkim.sock
+        non_smtpd_milters = local:/opendkim/opendkim.sock
+        ~~~
+
+    You can put this anywhere in the file. The usual practice is to put it after the `smtpd_recipient_restrictions` entry. You'll notice the path to the socket isn't the same here as it was in the `/etc/defaults/opendkim` file. That's because of Postfix's chroot jail, the path here is the path within that restricted view of the filesystem instead of within the actual filesystem.
+
+4.  Restart the OpenDKIM daemon so it sets up the correct socket for Postfix:
+
+        systemctl restart opendkim
+
+5.  Restart Postfix so it starts using OpenDKIM when processing mail:
+
+        systemctl restart postfix
+
+### Verify that everything's fully operational
+
+The easiest way to verify that everything's working is to send a test e-mail to `check-auth@verifier.port25.com` using an email client configured to submit mail to the submission port on your mail server. It will analyze your message and mail you a report indicating whether your email was signed correctly or not. It also reports on a number of other things such as SPF configuration and SpamAssassin flagging of your domain. If there's a problem, it'll report what the problem was.
+
+### Set up Author Domain Signing Practices (ADSP) (optional)
+
+As a final item, you can add an ADSP policy to your domain saying that all emails from your domain should be DKIM-signed. As usual, it's done with a TXT record for host `_adsp._domainkey` in your domain with a value of `dkim=all`. If you're using Linode's DNS Manager, the screen for the new text record will look like this:
+
+![Linode DNS manager add TXT record](/docs/assets/Postfix_ADSP_TXT_record.png)
+
+You don't need to set this up, but doing so makes it harder for anyone to forge email from your domains because recipient mail servers will see the lack of a DKIM signature and reject the message.
+
+### Key rotation
+
+The reason the YYYYMM format is used for the selector is that best practice calls for changing the DKIM signing keys every so often (monthly is recommended, and no longer than every 6 months). To do that without disrupting messages in transit, you generate the new keys using a new selector. The process is:
+
+1.  Generate new keys as in step 6 of "Configuring OpenDKIM". Do this in a scratch directory, not directly in `/etc/opendkim/keys`. Use the current year and month for the YYYYMM selector value, so it's different from the selector currently in use.
+
+2.  Use the newly-generated `.txt` files to add the new keys to DNS as in the DKIM "Setting Up DNS" section, using the new YYYYMM selector in the host names. Don't remove or alter the existing DKIM TXT records.
```

Comment:
I know it's a little late to think of this, but would it be a good idea at this point to instruct the reader to use the verify command before continuing to the next step and replacing the keys?